src/lib/timers/ingress_drop_monitor.lua

```diff
@@ -5,45 +5,59 @@ module(...,package.seeall)
 local ffi = require("ffi")
 
 -- Every 100 milliseconds.
-local interval = 1e8
+local default_interval = 1e8
 
-local with_restart = core.app.with_restart
 local now = core.app.now
 
-ingress_drop_monitor = {
-   threshold = 100000,
-   wait = 20,
-   last_flush = 0,
-   last_value = ffi.new('uint64_t[1]'),
-   current_value = ffi.new('uint64_t[1]'),
-}
+local IngressDropMonitor = {}
+
+function new(args)
+   local ret = {
+      threshold = args.threshold or 100000,
+      wait = args.wait or 20,
+      action = args.action or 'flush',
+      last_flush = 0,
+      last_value = ffi.new('uint64_t[1]'),
+      current_value = ffi.new('uint64_t[1]')
+   }
+   return setmetatable(ret, {__index=IngressDropMonitor})
+end
 
-function ingress_drop_monitor:sample ()
+function IngressDropMonitor:sample ()
    local app_array = engine.app_array
    local sum = self.current_value
    sum[0] = 0
    for i = 1, #app_array do
       local app = app_array[i]
       if app.ingress_packet_drops and not app.dead then
-         local status, value = with_restart(app, app.ingress_packet_drops)
-         if status then sum[0] = sum[0] + value end
+         sum[0] = sum[0] + app:ingress_packet_drops()
       end
    end
 end
 
-function ingress_drop_monitor:jit_flush_if_needed ()
+local tips_url = "https://github.com/Igalia/snabb/blob/lwaftr/src/program/lwaftr/doc/README.performance.md"
```

Comment:
Could we do this in a way that is more general? This code is being merged onto other branches and called from other programs now.